test/index.js

```diff
@@ -393,6 +394,80 @@ describe('scheme', () => {
         expect(res2.statusCode).to.equal(401);
     });
 
+    it('uauthorized error in validation function fails over to subsequent authentication scheme', async () => {
+
+        const plugin = {
+            name: 'bogusAuth',
+            register: (server, options) => {
+
+                const schema = () => {
+
+                    return {
+                        authenticate: (request, reply) => {
+
+                            return reply.authenticated({ credentials: { user: 'bogus-user' } });
+                        }
+                    };
+                };
+                server.auth.scheme('bogus', schema);
+            }
+        };
+
+        const server = Hapi.server();
+        await server.register(require('../'));
+        await server.register(plugin);
+        server.auth.strategy('first', 'cookie', {
+            password: 'password-should-be-32-characters',
+            ttl: 60 * 1000,
+            cookie: 'first',
+            validateFunc: function (request, session) {
+
+                throw Boom.unauthorized(null, 'first');
+            }
+        });
+
+        server.auth.strategy('second', 'bogus');
+
+        server.route({
+            method: 'GET', path: '/login/{user}',
+            config: {
+                handler: function (request, reply) {
+
+                    request.cookieAuth.set({ user: request.params.user });
+                    return reply.response(request.params.user);
+                }
+            }
+        });
+
+        server.route({
+            method: 'GET', path: '/resource',
+            config: {
+                auth: { mode: 'required', strategies: ['first', 'second'] },
+                handler: function (request, reply) {
+
+                    return reply.response('valid-resource');
+                }
+            }
+        });
+
+        const res = await server.inject('/login/bob');
+
+        expect(res.result).to.equal('bob');
+        const header = res.headers['set-cookie'];
+        expect(header.length).to.equal(1);
+        expect(header[0]).to.contain('Max-Age=60');
+        const cookie = header[0].match(/(?:[^\x00-\x20\(\)<>@\,;\:\\"\/\[\]\?\=\{\}\x7F]+)\s*=\s*(?:([^\x00-\x20\"\,\;\\\x7F]*))/);
+
+        server.inject({ method: 'GET', url: '/resource', headers: { cookie: 'first=' + cookie[1] } }, (res2) => {
```

Comment:
Can you add consistency to the way you inject requests? (i.e. use await)

Reply:
@mrlannigan change pushed.